dropwizard-client/src/main/java/io/dropwizard/client/DropwizardApacheConnector.java

```diff
@@ -0,0 +1,273 @@
+     * know the content length beforehand.
+     */
+    private static class JerseyRequestHttpEntity extends AbstractHttpEntity {
+
+        private ClientRequest clientRequest;
+
+        private JerseyRequestHttpEntity(ClientRequest clientRequest) {
+            this.clientRequest = clientRequest;
+            setChunked(true);
+        }
+
+        @Override
+        public boolean isRepeatable() {
+            return false;
+        }
+
+        @Override
+        public long getContentLength() {
+            return -1;
+        }
+
+        @Override
+        public InputStream getContent() throws IOException, IllegalStateException {
+            // Should never be called
+            return null;
+        }
+
+        @Override
+        public void writeTo(final OutputStream outputStream) throws IOException {
+            clientRequest.setStreamProvider(new OutboundMessageContext.StreamProvider() {
+                @Override
+                public OutputStream getOutputStream(int contentLength) throws IOException {
+                    return outputStream;
+                }
+            });
+            clientRequest.writeEntity();
+        }
+
+        @Override
+        public boolean isStreaming() {
+            return false;
+        }
+
+    }
+
+    /**
+     * A custom {@link org.apache.http.entity.AbstractHttpEntity} that uses
+     * a Jersey request as a content source.
+     * <p/>
+     * In contrast with {@link io.dropwizard.client.DropwizardApacheConnector.JerseyRequestHttpEntity},
+     * it's buffered. We preliminarily dump the content to a buffer before processing.
+     */
+    private static class BufferedJerseyRequestHttpEntity extends AbstractHttpEntity {
+
+        private static final int BUFFER_INITIAL_SIZE = 512;
+        private byte[] buffer;
+
+        private BufferedJerseyRequestHttpEntity(ClientRequest clientRequest) {
+            final ByteArrayOutputStream stream = new ByteArrayOutputStream(BUFFER_INITIAL_SIZE);
+            clientRequest.setStreamProvider(new OutboundMessageContext.StreamProvider() {
+                @Override
+                public OutputStream getOutputStream(int contentLength) throws IOException {
+                    return stream;
+                }
+            });
+            try {
+                clientRequest.writeEntity();
+            } catch (IOException e) {
+                throw new ProcessingException(LocalizationMessages.ERROR_BUFFERING_ENTITY(), e);
+            }
+            buffer = stream.toByteArray();
+            setChunked(false);
+        }
+
+        @Override
+        public boolean isRepeatable() {
+            return true;
+        }
+
+        @Override
+        public long getContentLength() {
+            return buffer.length;
+        }
+
+        @Override
+        public InputStream getContent() throws IOException, IllegalStateException {
+            // Shouldn't be called
+            return null;
```

Comment:
Agree with that. 

It's better to plainly  throw an exception to indicate that we don't 
support this method at all.